e2e/e2eTestRunnerWithCache.php

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env php
+<?php
+
+// runs a rector e2e test.
+// checks whether we expect a certain output, or alternatively that rector just processed everything without errors
+
+use Rector\Core\Console\Formatter\ColorConsoleDiffFormatter;
+use Rector\Core\Console\Formatter\ConsoleDiffer;
+use Rector\Core\Console\Style\SymfonyStyleFactory;
+use Rector\Core\Util\Reflection\PrivatesAccessor;
+use SebastianBergmann\Diff\Differ;
+use Symfony\Component\Console\Command\Command;
+
+$projectRoot = __DIR__ .'/..';
+$rectorBin = $projectRoot . '/bin/rector';
+$autoloadFile = $projectRoot . '/vendor/autoload.php';
+
+// so we can use helper classes here
+require_once __DIR__ . '/../vendor/autoload.php';
+
+$e2eCommand = 'php '. $rectorBin .' process --dry-run --no-ansi -a '. $autoloadFile;
```

Comment:
This runner is a lighter version of the original one, removing the `--clear-cache` here, and support for custom parameters because they are not needed for now